Fix/playwright tags

AZANIR opened the pull request.

AZANIR committed:
Add tag extraction functionality to Playwright reporter

- Implemented `extractTags` function to normalize and deduplicate tags from test titles, options, and parent suites.
- Updated Playwright reporter to utilize extracted tags in test run payload.
- Added comprehensive tests for tag extraction covering various scenarios, including title tags, options tags, and inheritance from parent suites.
- Updated `.gitignore` to include `.claude/` directory.

AZANIR committed:
Add tags parameter to Client class methods

- Updated the Client class to include a `tags` parameter in the relevant methods, enhancing the ability to manage and utilize tags in test data.
- This change aligns with recent improvements to tag extraction functionality in the Playwright reporter.

AZANIR committed:
Merge branch '2.x' of https://github.com/testomatio/reporter into fix/playwright-tags

AZANIR committed:
Enhance test configuration and debug file handling

- Added `apiKey` configuration for Testomatio in CodeceptJS setup.
- Updated Playwright reporter path to use the correct adapter location.
- Improved debug file retrieval logic in Codecept test runner to find the most recent debug file instead of relying on symlink, ensuring more reliable debugging.
- Adjusted formatting in hook steps to match regular step format for consistency.
- Updated test cases to use the correct tag format for better organization and clarity.

olexandr13 commented on src/adapter/codecept.js:
```diff
@@ -461,13 +461,14 @@ function formatCodeceptStep(step) {
   let title = step.name;
   if (step.actor && step.name) {
-    title = `${step.actor}.${step.name}`;
+    // Use space notation to match regular CodeceptJS step format (I expectEqual instead of I.expectEqual)
```
🤔

olexandr13 commented on tests/unit/playwright_tags_test.js:
```diff
@@ -0,0 +1,279 @@
+
+// Import the extractTags function from the adapter
+// Note: We need to mock the test object structure since it's a private function
+function extractTags(test) {
```
no need to duplicate function declaration here
import from adapter

AZANIR committed:
Refactor Playwright adapter and improve tag extraction tests

- Updated Playwright reporter path to the correct source location.
- Removed redundant tag extraction logic from the `extractTags` function in the Playwright adapter.
- Simplified the test suite for tag extraction by directly importing the `extractTags` function, enhancing test clarity and maintainability.
